test/ERC7914.t.sol

```diff
@@ -359,4 +362,146 @@ contract ERC7914Test is DelegationHandler {
+        TestKey memory testKey = TestKeyManager.withSeed(KeyType.Secp256k1, 0x123456);
+        vm.prank(address(signerAccount));
+        signerAccount.register(testKey.toKey());
+        
+        // Create permit details with witness data
+        Permit2Utils.MockWitness memory witnessData = Permit2Utils.MockWitness(10000000, address(5), true);
+        bytes32 witness = keccak256(abi.encode(witnessData));
+        ISignatureTransfer.PermitTransferFrom memory permit = ISignatureTransfer.PermitTransferFrom({
+            permitted: ISignatureTransfer.TokenPermissions({
+                token: address(erc20Eth),
+                amount: spendAmount
+            }),
+            nonce: 0,
+            deadline: block.timestamp + 1 hours
+        });
+
+        // Get domain information for ERC7739 signing
+        bytes memory signerAccountDomainBytes = TypedDataSignBuilder.toDomainBytes(IERC5267(address(signerAccount)));
+        
+        // Create ERC7739 signature for permit2
+        bytes memory erc7739Sig = testKey.signPermitTransfer7739(
+            permit,
+            Permit2Utils.FULL_EXAMPLE_WITNESS_TYPEHASH,
+            witness,
+            permit2.DOMAIN_SEPARATOR(),
+            signerAccountDomainBytes,
+            bob
+        );
+        
+        // Wrap the ERC7739 signature for the signer account
+        bytes memory wrappedSignature = abi.encode(testKey.toKeyHash(), erc7739Sig, "");
+
+        // Bob cannot transfer more than the approved amount
+        ISignatureTransfer.SignatureTransferDetails memory invalidTransferDetails = ISignatureTransfer.SignatureTransferDetails({
+            to: bob,
+            requestedAmount: spendAmount + 1
+        });
+
+        vm.expectRevert(abi.encodeWithSelector(ISignatureTransfer.InvalidAmount.selector, spendAmount));
+        vm.prank(bob);
+        permit2.permitWitnessTransferFrom(permit, invalidTransferDetails, address(signerAccount), witness, Permit2Utils.WITNESS_TYPE_STRING, wrappedSignature);
+
+        // Bob can transfer the approved amount
+        ISignatureTransfer.SignatureTransferDetails memory validTransferDetails = ISignatureTransfer.SignatureTransferDetails({
+            to: bob,
+            requestedAmount: spendAmount
+        });
+        vm.prank(bob);
+        permit2.permitWitnessTransferFrom(permit, validTransferDetails, address(signerAccount), witness, Permit2Utils.WITNESS_TYPE_STRING, wrappedSignature);
+
+        // Verify the transfer
+        assertEq(bob.balance, spendAmount);
+        assertEq(address(signerAccount).balance, totalAmount - spendAmount);
+    }
+
+    // Test that a permit2 signature from a 7739 signer can be used to
+    // transfer native ETH using the ERC20-eth contract (no witness)
+    function test_permit2SignatureTransferNativeWith7739() public {
+        // Deploy ERC20ETH and Permit2
+        ERC20ETH erc20Eth = new ERC20ETH();
+        IPermit2 permit2 = IPermit2(Permit2Utils.deployPermit2());
```

Comment:
When you deploy permit2 like this from bytecode what is the chainId? It is immutable (so in the deployment bytecode), which affects the domain separator.

Reply:
It looks like Permit2 (via [EIP712](https://github.com/Uniswap/permit2/blob/cc56ad0f3439c502c246fc5cfcc3db92bb8b7219/src/EIP712.sol#L20)) sets the chainId based on block.chainId so it should be dependent upon where it's deployed. LMK if I misunderstood your question.

Reply:
Right but I think when you deploy it like this via bytecode, since immutables are part of the bytecode, it must include a chainId already. I'm just not sure which chain id haha

Reply:
Because the value it's immutable (not constant) and is assigned within the constructor, it pulls it at deployment time (not compile time).
https://docs.soliditylang.org/en/latest/contracts.html#constant-and-immutable-state-variables
> for constant variables, the value has to be fixed at compile-time, while for immutable, it can still be assigned at construction time.